implementations/rust/ockam/ockam_entity/src/entity.rs

```diff
@@ -105,7 +110,7 @@ impl Entity {
 impl ProfileAdd for Entity {
     fn add_profile(&mut self, profile: ProfileSync) -> Result<()> {
         if let Ok(id) = profile.identifier() {
-            if let Some(_) = self.profiles.insert(id, profile) {
+            if self.profiles.insert(id, profile).is_some() {
```

Comment:
Did you consider writing such things like this?
```rs
let id = profile.identifier().or(InvalidInternalState.into())?;
let _ = self.profiles.insert(id, profile).ok_or_else(InvalidInternalState.into())?;

Ok(())
```
Do you think it looks cleaner?

Reply:
The downside is duplicating Errors, but you reduce code nesting, which is nice

Reply:
Yeah I think I like this more, the nesting is pretty annoying